components/dashboard/src/configurations/detail/ConfigurationDeleteModal.tsx

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import type { Project } from "@gitpod/gitpod-protocol";
+import { FunctionComponent, useCallback, useState } from "react";
+import ConfirmationModal from "../../components/ConfirmationModal";
+import { useDeleteProject } from "../../data/projects/project-queries";
+
+type RemoveConfigurationModalProps = {
+    configuration: Project;
+    onClose: () => void;
+    onRemoved: () => void;
+};
+
+export const RemoveConfigurationModal: FunctionComponent<RemoveConfigurationModalProps> = ({
+    configuration,
+    onClose,
+    onRemoved,
+}) => {
+    const [disabled, setDisabled] = useState(false);
+    const useDeleteRepository = useDeleteProject();
+
+    const removeConfiguration = useCallback(async () => {
+        setDisabled(true);
```

Comment:
We shouldn't need to track this state, we can use the query `isLoading` prop for this instead.